datafusion/spark/src/function/datetime/date_add.rs

```diff
@@ -136,3 +145,48 @@ fn spark_date_add(args: &[ArrayRef]) -> Result<ArrayRef> {
     };
     Ok(Arc::new(result))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::datatypes::Field;
+
+    #[test]
+    fn test_date_add_non_nullable_inputs() {
+        let func = SparkDateAdd::new();
+        let args = &[
+            Arc::new(Field::new("arg_int8", DataType::Int8, false)),
+            Arc::new(Field::new("arg_int16", DataType::Int16, false)),
+        ];
+
+        let ret_field = func
+            .return_field_from_args(ReturnFieldArgs {
+                arg_fields: args,
+                scalar_arguments: &[None, None],
+            })
+            .unwrap();
+
+        assert_eq!(ret_field.data_type(), &DataType::Date32);
+        assert!(!ret_field.is_nullable());
+    }
+
+    #[test]
+    fn test_date_add_nullable_inputs() {
+        let func = SparkDateAdd::new();
+        let args = &[
+            Arc::new(Field::new("arg_int8", DataType::Int8, false)),
```

Comment:
The possible signatures (https://github.com/apache/datafusion/pull/19229/files#diff-8b0a95c4b80c4d83fe7c0752f11a762647f4d36173ae7988a76d7bc7da5d5168R53-R55) say that the first argument should be `Date32`
```suggestion
            Arc::new(Field::new("arg_date32", DataType::Date32, false)),
```